### PR TITLE
KOGITO-124: [DMN Designer] Cannot define Data Type Range constraint

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/KOGITO-124

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/FEELRangeParserTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+public class FEELRangeParserTest {
+
+    @Parameterized.Parameters(name = "{index}: input={0}, expected={1}")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {null, new RangeValue()},
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/FEELRangeParserTest.java]: Please implement `equals` and `hashCode` of `org.kie.workbench.common.dmn.api.editors.types.RangeValue`

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/FEELRangeParser.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    private static final String EXCLUDE_START = "(";
+    private static final String INCLUDE_END = "]";
+    private static final String EXCLUDE_END = ")";
+
+    public static RangeValue parse(final String input) {
+        final RangeValue rangeValue = new RangeValue();
+        if (Objects.isNull(input)) {
+            return rangeValue;
+        }
+        final String _input = input.trim();
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/main/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/FEELRangeParser.java]: Updated.

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/FEELRangeParserTest.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-common/src/test/java/org/kie/workbench/common/dmn/webapp/kogito/common/client/services/FEELRangeParserTest.java]: Done and test improved (to make use of new `equals(..)` method on `RangeValue`).

Also, your PR's new tests revealed some bugs so fixed those too. Great stuff.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**